test/helpers/kubectl.go

```diff
@@ -899,7 +899,17 @@ func (kub *Kubectl) GatherLogs() {
 		"kubectl describe pods --all-namespaces -o json":             "pods_status.txt",
 		"kubectl get replicationcontroller --all-namespaces -o json": "replicationcontroller.txt",
 		"kubectl get deployment --all-namespaces -o json":            "deployment.txt",
-		"kubectl -n kube-system logs -l k8s-app=cilium --timestamps": "cilium_logs.txt",
+	}
+
+	ciliumPods, err := kub.GetCiliumPods(KubeSystemNamespace)
+	if err != nil {
+		kub.logger.WithError(err).Error("Cannot get cilium pods")
+	}
+
+	for _, pod := range ciliumPods {
+		key := fmt.Sprintf("kubectl -n kube-system logs --timestamps %s", pod)
```

Comment:
Does it make sense to also get the `previous`, or not?

Reply:
Can't hurt IMO. We can follow that up separately because the sooner we get this PR merged, the better; lack of this functionality in master has already made it difficult to triage build failures. 

Created https://github.com/cilium/cilium/issues/3654 for tracking that request.